### PR TITLE
feat(components/state): state key contains characters ||

### PR DESCRIPTION
**Pull request by 1046102779**
state key includes keyPrefix and user's request key that both maybe contains illegal characters '||'

keyPrefix: a||b

user request key: a||b

### CODE REVIEW

**Review by artursouza — DISMISSED** [review]

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -840,7 +840,11 @@ func (a *DaprRuntime) initState(s components_v1alpha1.Component) error {
 		}
 
 		a.stateStores[s.ObjectMeta.Name] = store
-		state_loader.SaveStateConfiguration(s.ObjectMeta.Name, props)
+		err = state_loader.SaveStateConfiguration(s.ObjectMeta.Name, props)
+		if err != nil {
+			diag.DefaultMonitoring.ComponentInitFailed(s.Spec.Type, "init")
+			return err
```
Comment (mukundansundar) [inline pkg/runtime/runtime.go:846]: add a Warnf log here?

**Review by mukundansundar — COMMENTED** [review]
@wcs1only @artursouza Should there be consistency in terms of error states between HTTP API and gRPC APIs? In the sense if an error is returned in gRPC, shouldn't an error response be returned from HTTP side as well instead of debug logging the error? 

In this case I think the error should be ERR_MALFORMED_REQUEST ? Any thoughts?

File: pkg/http/api.go [hunk 2]
```diff
@@ -480,8 +493,13 @@ func (a *api) onGetState(reqCtx *fasthttp.RequestCtx) {
 
 	key := reqCtx.UserValue(stateKeyParam).(string)
 	consistency := string(reqCtx.QueryArgs().Peek(consistencyParam))
+	k, err := state_loader.GetModifiedStateKey(key, storeName, a.id)
+	if err != nil {
+		log.Debug(err)
+		return
```
[the remark this answers is not in the document]
Comment (artursouza) [reply pkg/http/api.go:499]: errors should be returned to the user and not swallowed.

**Comment by artursouza** [comment]
> @wcs1only @artursouza Should there be consistency in terms of error states between HTTP API and gRPC APIs? In the sense if an error is returned in gRPC, shouldn't an error response be returned from HTTP side as well instead of debug logging the error?
> 
> In this case I think the error should be ERR_MALFORMED_REQUEST ? Any thoughts?

Behavior between GRPC and HTTP should be the same regarding error handling: if error is returned in one, the the same must be done in the other. On the other hand, the error codes themselves will not be identical since GRPC and HTTP have differences in the how errors are exposed.

**Comment by 1046102779** [comment]
@mukundansundar @artursouza please review PR

**Review by artursouza — APPROVED** [review]

**Pull request merged**